applications/README.md

```diff
@@ -0,0 +1,488 @@
+# AICA Applications
+
+An AICA application is interconnected graph of **components** and **hardware interfaces**. Components process data in a
+periodic step function and transfer data as signals to other components and hardware interfaces. Hardware interfaces
+are drivers that connect controllers to robots. The application state can be dynamically changed through **events**,
+which can change parameters, load or unload components, trigger specific service actions and more.
+
+The nodes and edges of an AICA application graph can be visualized diagrammatically. It is also possible to define the
+graph structure in a text format using YAML syntax.
+
+---
+
+# The YAML Application
+
+The following sections define the YAML application syntax used to describe an AICA application.
+
+- [Overview](#overview)
+- [On Start](#on-start)
+- [Hardware Interfaces](#hardware-interfaces)
+    - [URDF](#urdf)
+    - [Rate](#rate)
+    - [Controllers](#controllers)
+- [Conditions](#conditions)
+  - [Simple Conditions](#simple-conditions)
+  - [Conditional Operators](#conditional-operators)
+  - [Nested Conditions](#nested-conditions)
+- [Components](#components)
+    - [Component](#component)
+    - [Mapping](#mapping)
+    - [Log Level](#log-level)
+    - [Parameters](#parameters)
+    - [Events](#events)
+- [Validating a YAML Application](#validating-a-yaml-application)
+
+## Overview
+
+An application description contains the following top-level elements.
+
+```yaml
+on_start:
+  ...
+
+hardware_interfaces:
+  ...
+
+conditions:
+  ...
+
+components:
+  ...
+```
+
+
+## On Start
+
+The `on_start` keyword is reserved as a special event trigger when the application is launched.
+List the [events](#events) to trigger on startup (for example, to load components).
+
+```yaml
+on_start:
+  load:
```

Comment:
You know, now that I read this again, and we are renaming state engine methods, wouldn't it be more correct to have `load_component` and `unload_component`. To a user it might seem unclear why components are somehow more important than controllers and hardware

Reply:
On second thought, I think its okay, because otherwise `set`, `service`, and `lifecycle` would almost need a more verbose name too

Reply:
It's a fair point but I think component-centric events are the default so it's nice to have the brevity. As long as it's well documented I think it's ok like this.